client/client-lib/src/mint/backup.rs

```diff
@@ -246,18 +246,17 @@ impl MintClient {
         backup: PlaintextEcashBackup,
         gap_limit: usize,
     ) -> Result<Cancellable<EcashRecoveryFinalState>> {
-        let end_epoch = match self.context.api.fetch_last_epoch().await {
+        let current_epoch_count = match self.context.api.fetch_epoch_count().await {
             Ok(v) => v,
             Err(e) => {
                 return Err(e.into());
             }
         };
-        let epoch_range = backup.epoch..=end_epoch;
+        // TODO: This -1 is probably not necessary, as it should be enough to start from the exact epoch the snapshot was taken, but it is harmless to start from any epoch in the past, and starting a bit earlier makes it more robust in face of some inconsistency that we've missed.
```

Comment:
Is this claim true that double-processing of epochs is harmless @dpc?

Reply:
Yes. Just like you can always start restore from zero, you could start it from 1, or any other number in between. Restore code looks for things it expects and ignores everything else.

Reply:
Would be good to just have a test to see if we need the sub or not (comment @mxz42).